src/main/kotlin/org/jetbrains/teamcity/rest/api.kt

```diff
@@ -181,3 +183,15 @@ interface TriggeredInfo {
     val user: User?
     val build: Build?
 }
+
+interface FinishBuildTrigger {
+    val initiatedBuildConfiguration: BuildConfigurationId
+    val afterSuccessfulBuildOnly: Boolean
+    val branchFilter: List<BranchRule>
+}
+
+interface BranchRule {
+    val rule: String
```

Comment:
Why do you need to expose unparsed rule string into API? All the information is in `include` and `name` branches, isn't it?
Excessive elements in API just makes it harder to understand